addons/block_code/ui/util.gd

```diff
@@ -12,3 +12,11 @@ static func node_is_part_of_edited_scene(node: Node) -> bool:
 
 	var edited_scene_parent := tree.edited_scene_root.get_parent()
 	return edited_scene_parent and edited_scene_parent.is_ancestor_of(node)
+
+
+## Get the nearest Block node that is a parent of the provided node.
+static func get_parent_block(node: Node) -> Block:
```

Comment:
I ended up putting this in the new `BlockTreeUtil` class I made so as not to re-introduce a circular dependency. The `Util` class otherwise depends only on native classes.